modules/zip/src/zip-loader.ts

```diff
@@ -17,52 +22,24 @@ export const ZipLoader: LoaderWithParser<FileMap, never, LoaderOptions> = {
   mimeTypes: ['application/zip'],
   category: 'archive',
   tests: ['PK'],
-  options: {},
+  options: {
+    zip: {
+      mode: 'all'
+    }
+  },
   parse: parseZipAsync
 };
 
-// TODO - Could return a map of promises, perhaps as an option...
-async function parseZipAsync(data: any, options = {}): Promise<FileMap> {
-  const promises: Promise<any>[] = [];
-  const fileMap: Record<string, ArrayBuffer> = {};
-
-  try {
-    const jsZip = new JSZip();
-
-    const zip = await jsZip.loadAsync(data, options);
-
-    // start to load each file in this zip
-    zip.forEach((relativePath, zipEntry) => {
-      const subFilename = zipEntry.name;
-
-      const promise = loadZipEntry(jsZip, subFilename, options).then((arrayBufferOrError) => {
-        fileMap[relativePath] = arrayBufferOrError;
-      });
-
-      // Ensure Promise.all doesn't ignore rejected promises.
-      promises.push(promise);
-    });
-
-    await Promise.all(promises);
-    return fileMap;
-  } catch (error) {
-    // @ts-ignore
-    options.log.error(`Unable to read zip archive: ${error}`);
-    throw error;
-  }
-}
-
-async function loadZipEntry(jsZip: any, subFilename: string, options: any = {}) {
-  // jszip supports both arraybuffer and text, the main loaders.gl types
-  // https://stuk.github.io/jszip/documentation/api_zipobject/async.html
-  try {
-    const arrayBuffer = await jsZip.file(subFilename).async(options.dataType || 'arraybuffer');
-    return arrayBuffer;
-  } catch (error) {
-    options.log.error(`Unable to read ${subFilename} from zip archive: ${error}`);
-    // Store error in place of data in map
-    return error;
+async function parseZipAsync(data: ArrayBuffer, options?: ZipLoaderOptions) {
+  switch (options?.zip?.mode) {
```

Comment:
Nit: For another PR. This creates a loader with two very different return types. If supporting this is important, think we should consider splitting into two loaders.

Reply:
Normally when reading single files I would expect users to work with the ZipFileSystem